specification/programmableconnectivity/resource-manager/Microsoft.ProgrammableConnectivity/preview/2023-11-01-preview/OpenApiGateways.json

```diff
@@ -0,0 +1,745 @@
+        "x-ms-examples": {
+          "OpenApiGateways_UpdateCredentials_MaximumSet_Gen": {
+            "$ref": "./examples/OpenApiGateways_UpdateCredentials_MaximumSet_Gen.json"
+          },
+          "OpenApiGateways_UpdateCredentials_MinimumSet_Gen": {
+            "$ref": "./examples/OpenApiGateways_UpdateCredentials_MinimumSet_Gen.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AccountType": {
+      "type": "string",
+      "description": "The Account Type of the Open API Gateways",
+      "enum": [
+        "AzureManaged",
+        "UserManaged"
+      ],
+      "x-ms-enum": {
+        "name": "AccountType",
+        "modelAsString": true,
+        "values": [
+          {
+            "name": "AzureManaged",
+            "value": "AzureManaged",
+            "description": "Managed by Azure on-behalf-of the user"
+          },
+          {
+            "name": "UserManaged",
+            "value": "UserManaged",
+            "description": "Managed by the User themselves on the Operator end"
+          }
+        ]
+      }
+    },
+    "AppCredentials": {
+      "type": "object",
+      "description": "Application credentials used to call the operator's Network APIs",
+      "properties": {
+        "appId": {
+          "type": "string",
+          "description": "Application ID"
+        },
+        "appSecret": {
+          "type": "string",
+          "format": "password",
+          "description": "Application Secret",
+          "x-ms-secret": true
+        }
+      },
+      "required": [
+        "appId",
+        "appSecret"
+      ]
+    },
+    "OpenApiGateway": {
+      "type": "object",
+      "description": "A Programmable Connectivity Open API Gateway resource",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/OpenApiGatewayProperties",
+          "description": "The resource-specific properties for this resource.",
+          "x-ms-client-flatten": true,
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource"
+        }
+      ]
+    },
+    "OpenApiGatewayListResult": {
+      "type": "object",
+      "description": "The response of a OpenApiGateway list operation.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The OpenApiGateway items on this page",
+          "items": {
+            "$ref": "#/definitions/OpenApiGateway"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "OpenApiGatewayProperties": {
+      "type": "object",
+      "description": "APC Gateway properties",
+      "properties": {
+        "gatewayId": {
```

Comment:
Is it an azure resource ID? If so, please add "format": "arm-id" and "x-ms-arm-id-details".

https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details